src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1424,6 +1484,107 @@ protected void GenerateInitTrackCount()
             Ret();
         }
 
+        /// <summary>Computes a min bound on the required length of any string that could possibly match.</summary>
+        /// <returns>The min computed length.  If the result is 0, there is no minimum we can enforce.</returns>
+        protected int ComputeMinLength(RegexNode node, int depth = 0)
```

Comment:
Should this be properly on the RegexNode class?

Reply:
Yes, I have that locally, as part of https://github.com/dotnet/runtime/pull/1624#issuecomment-573379306.

Reply:
Also , this might be worth testing through reflection - thoughts? It would be more reassuring than relying on the existing test cases to discover it from outside in . It seems a well defined method to test.

Reply:
I'd rather not.  It's impact can be pretty easily validated via regexes.  Why is that concerning?

Reply:
Fair enough. I guess I'm just worried about yet more branches to test - might we have test regexes that will no longer exercise what they did before, as they now fall out of this fast path?

Reply:
> as they now fall out of this fast path?

It's possible, but for that to happen the inputs would never have been able to match; they'd get part of the way through the pattern, and then there'd be nothing left.  Regardless of this change, we should ensure we're testing our regexes against inputs that are too short, exactly right, and longer than needed, and this change will just make some of the "too short" ones execute faster.